src/directives/schema-validate.js

```diff
@@ -1,110 +1,173 @@
+            ngModel.$setValidity('tv4-' + result.error.code, false);
+            error = result.error;
+            return undefined;
+          }
+          return viewValue;
+        };
+
+        // Custom validators, parsers, formatters etc
+        if (typeof form.ngModel === 'function') {
+          form.ngModel(ngModel);
         }
-        return viewValue;
-      };
 
-      // Custom validators, parsers, formatters etc
-      if (typeof form.ngModel === 'function') {
-        form.ngModel(ngModel);
-      }
+        ['$parsers', '$viewChangeListeners', '$formatters'].forEach(function(attr) {
+          if (form[attr] && ngModel[attr]) {
+            form[attr].forEach(function(fn) {
+              ngModel[attr].push(fn);
+            });
+          }
+        });
 
-      ['$parsers', '$viewChangeListeners', '$formatters'].forEach(function(attr) {
-        if (form[attr] && ngModel[attr]) {
-          form[attr].forEach(function(fn) {
-            ngModel[attr].push(fn);
-          });
-        }
-      });
+        ['$validators', '$asyncValidators'].forEach(function(attr) {
+          // Check if our version of angular has i, i.e. 1.3+
+          if (form[attr] && ngModel[attr]) {
+            angular.forEach(form[attr], function(fn, name) {
+              ngModel[attr][name] = fn;
+            });
+          }
+        });
 
-      ['$validators', '$asyncValidators'].forEach(function(attr) {
-        // Check if our version of angular has i, i.e. 1.3+
-        if (form[attr] && ngModel[attr]) {
-          angular.forEach(form[attr], function(fn, name) {
-            ngModel[attr][name] = fn;
-          });
-        }
-      });
-
-      // Get in last of the parses so the parsed value has the correct type.
-      // We don't use $validators since we like to set different errors depeding tv4 error codes
-      ngModel.$parsers.push(validate);
-
-      // Listen to an event so we can validate the input on request
-      scope.$on('schemaFormValidate', function() {
-        if (ngModel.$setDirty) {
-          // Angular 1.3+
-          ngModel.$setDirty();
-          validate(ngModel.$modelValue);
-        } else {
-          // Angular 1.2
-          ngModel.$setViewValue(ngModel.$viewValue);
+        // Get in last of the parses so the parsed value has the correct type.
+        // We don't use $validators since we like to set different errors depeding tv4 error codes
+        ngModel.$parsers.push(validate);
+
+        // Listen to an event so we can validate the input on request
+        scope.$on('schemaFormValidate', function() {
+          if (ngModel.$setDirty) {
+            // Angular 1.3+
+            ngModel.$setDirty();
+            validate(ngModel.$modelValue);
+          } else {
+            // Angular 1.2
+            ngModel.$setViewValue(ngModel.$viewValue);
+          }
+
+        });
+
+
+        var DEFAULT_DESTROY_STRATEGY;
+        if (scope.options && scope.options.formDefaults) {
```

Comment:
`formDefaults` should not be checked directly, it's already copied to the form object, see https://github.com/Textalk/angular-schema-form/blob/development/src/services/schema-form.js#L70

But in this case with `null` as a valid option distinct from `undefined` it might be better to move this to a global option. i.e. instead of checking `scope.option.formDefaults.destroyStrategy` just check `scope.option.destroyStrategy`